core/services/gateway/network/httpclient.go

```diff
@@ -0,0 +1,105 @@
+package network
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+// HttpClient interfaces defines a method to send HTTP requests
+// handles retries and timeouts
+type HttpClient interface {
+	Send(ctx context.Context, req HttpRequest) (*HttpResponse, error)
+}
+
+type HttpRequest struct {
+	Method     string
+	URL        string
+	Headers    map[string]string
+	Body       []byte
+	Timeout    time.Duration
+	RetryCount uint8
+}
+
+type HttpResponse struct {
+	StatusCode int               // HTTP status code
+	Headers    map[string]string // HTTP headers
+	Body       []byte            // Base64-encoded binary body
+}
+
+type httpClient struct {
+	client *http.Client
+	lggr   logger.Logger
+}
+
+func NewHttpClient(tlsConfig *tls.Config, defaultTimeout time.Duration, lggr logger.Logger) (HttpClient, error) {
+	transport := http.DefaultTransport
+	if tlsConfig != nil {
```

Comment:
Likely won't be needed for an HTTPClient. This is useful for client-side certs, which we won't support. I think you can ignore this param for simplicity and just leave a comment saying it could be extended in the future.